k-alexey/ggrc-core
# GGRC-2365 Export snapshots archived column

# Issue description

'Archived' column added to export of snapshots

# Steps to test the changes

Create Audit with mapped controls, and make an export of snapshots. CSV file should contain 'Archived' column with archived status of related audit.

# Solution description

- Add 'archived' to export code base
- Add foreign key relationship to Snapshot-Audit to eagerly load archived status with constant number of queries.
- Fix broken tests(not expecting 'Archived' column)
- Add test of exporting snapshots with archived status

# Sanity checklist

*Change Snapshot.parent_id to foreign key to Audit.id*

# Migration checklist
*Migration will delete snapshots with wrong foreign key parent_id or parent_type not equal to 'Audit'*

# PR Review checklist
